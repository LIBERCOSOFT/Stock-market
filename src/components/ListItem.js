import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const ListItem = ({ stock }) => (
  <li>
    <Link to={`/details/${stock.symbol}`}>
      <div>
        <p>
          <span>Name:</span>
          {' '}
          {stock.companyName}
        </p>
        <p>
          <span>Industry:</span>
          {' '}
          {stock.industry}
        </p>
        <p>
          <span>Price per Stock:</span>
          {' '}
          $
          {stock.price}
        </p>
        <p>
          <span>Exchange:</span>
          {' '}
          {stock.exchange}
        </p>
        <p>
          <span>Market Cap:</span>
          {' '}
          $
          {stock.marketCap.toLocaleString()}
        </p>
        <p>
          <span>
            See about
            {' '}
            {stock.symbol}
          </span>
          {' '}
          <i className="fa-sharp fa-solid fa-arrow-right" />
        </p>
      </div>
    </Link>
  </li>
);

ListItem.propTypes = {
  stock: PropTypes.instanceOf(Object).isRequired,
};

export default ListItem;
