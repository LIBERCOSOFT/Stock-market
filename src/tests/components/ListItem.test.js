import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import ListItem from '../../components/ListItem';

test('renders list-all-stock component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <ListItem />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
