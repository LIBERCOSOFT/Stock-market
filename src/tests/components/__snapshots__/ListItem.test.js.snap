// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list-all-stock component snapshot 1`] = `
<DocumentFragment>
  <li>
    <a
      href="/details/"
    >
      <div>
        <p>
          <span>
            Name:
          </span>
           
        </p>
        <p>
          <span>
            Industry:
          </span>
           
        </p>
        <p>
          <span>
            Price per Stock:
          </span>
           $0
        </p>
        <p>
          <span>
            Exchange:
          </span>
           
        </p>
        <p>
          <span>
            Market Cap:
          </span>
           $0
        </p>
        <p>
          <span>
            See about 
          </span>
           
          <i
            class="fa-sharp fa-solid fa-arrow-right"
          />
        </p>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
